CONTRIBUTING.md

```diff
@@ -18,6 +18,16 @@ When filing an issue, please provide as much of the following information as pos
 
 We love contributions to Realm! If you'd like to contribute code, documentation, or any other improvements, please [file a Pull Request](https://github.com/realm/realm-core/pulls) on our GitHub repository. Make sure to accept our [CLA](#cla) and to follow our [style guide](doc/development/coding_style_guide.cpp).
 
+### Branches
+
+Most changes should always go in the `master` branch - this means bugfixes, small enhancements, and features that do not introduce "behavior-breaking" changes. The exception is any code that touches the file format or sync protocol, as well as non-trivial changes to APIs - those should target the `breaking` branch. When in doubt please get in touch with the [Core Database team](https://github.com/orgs/realm/teams/core-database).
+Occasionally a single PR, while a complete unit of work, does not contain everything needed to make a release. For example, adding a new data type involves work in the storage, query, schema, and sync layers which is usually shared by several contributors. In this case we use a feature integration branch. The current integration branch and its theme are announced with a pinned topic in [Discussions](https://github.com/realm/realm-core/discussions).
+
+### Changelog
+
+Pull requests that touch code should always add an entry to `CHANGELOG.md` with a sentence or two describing a change and a link to the relevant issue or the PR itself. The expectation is that the entry gives a general idea about the change, and then the PR or issue describes the nature of the change in more detail. Bug fixes should also mention the version the bug was introduced in.
+Changelog entries should be high-level enough that they make sense when viewed by users of the Realm SDKs. A good example is `Fixed an issue where opening a realm from two different processes on Windows results in a deadlock (since vX.Y.Z)` instead of `Fixed a bug with signaling win32 robust mutexes`. The nature of the change should be discussed in the PR or issue whereas the changelog entry describes the visible outcome to users. For anything else please use the `Internals` section of `CHANGELOG.md`.
```

Comment:
We also include a link to either the issue it solves or if no issue, the PR.

Reply:
Does't this say the same thing?
```
Pull requests that touch code should always add an entry to `CHANGELOG.md` with a sentence or two describing a change and a link to the relevant issue or the PR itself.
```

Reply:
it does. So perhaps just include it in the example?

Reply:
Thanks. I added a link in the example.